### PR TITLE
TST: Enable pytest-xdist

### PR DESCRIPTION
**Pull request by Kojoley**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -166,6 +163,12 @@ script:
           gdb -return-child-result -batch -ex r -ex bt --args python $PYTHON_ARGS tests.py $NOSE_ARGS $TEST_ARGS
         fi
       else
+        # Workaround for pytest-xdist flaky colletion order
+        # https://github.com/pytest-dev/pytest/issues/920
+        # https://github.com/pytest-dev/pytest/issues/1075
+        export PYTHONHASHSEED=$(shuf -i 1-4294967295 -n 1)
+        echo PYTHONHASHSEED=$PYTHONHASHSEED
```
Comment (QuLogic) [inline .travis.yml]: Will this affect the determinism tests?

**Comment by tacaswell** [comment]
It still looks longer, the nose version it matches is the one for coverage which does not use parallelism.  Am I confused?

File: .travis.yml (hunk 1 above)
Comment (tacaswell) [inline .travis.yml]: Aren't the tests that really care about this run in a shell-call with their own seeds set?

**Comment by QuLogic** [comment]
Oh, I wasn't looking at the parallelism, just that it wasn't exceptionally slow.

File: .travis.yml (hunk 1 above)
Comment (QuLogic) [inline .travis.yml]: I don't see it set/unset `PYTHONHASHSEED`, so wouldn't it be inherited?

Comment (tacaswell) [inline .travis.yml]: Yeah, I think you are right :sheep:

**Comment by Kojoley** [comment]
We do not run all the tests from nose. `test_axes` takes 2 mins on my machine with `-n 2`. IIRC nose builds previously had taken about 12 mins, what is close to the build time we see on pytest build here.

**Comment by Kojoley** [comment]
Also the slowdown may be related to the problem described here pytest-dev/pytest-xdist#18

File: .travis.yml (hunk 1 above)
Comment (Kojoley) [inline .travis.yml]: Well, I have find out that xdist collection problem is triggered by `test_mathtext.py`, but I do not know what exactly, maybe `globals()` injection magic or something else. I think the best option for us is to rewrite `test_mathtext` to pytest fixtures and forget about this xdist issue.

Comment (QuLogic) [inline .travis.yml]: Hmm, that definitely is weird; is there a way to parametrize the fileename passed to the `image_comparison` decorator?

**Comment by tacaswell** [comment]
sorry, I think I was too optimistic about merging this.

**Pull request merged**